### PR TITLE
More descriptive HTTP response codes + Swagger 2.0 to OpenAPI 3.0

### PR DESCRIPTION
**Pull request by geebee**
- DestinationDoesNotExist error changed from 400 (Bad Request) to 404
(Not Found)
  - DestinationExistsError changed from 400 (Bad Request) to 409
(Conflict)

NOTE: Because of the change in response codes (and therefore the API
contract with clients), the major version number has increased from 1 to
2 (new version number is 2.0.0)

NOTE: Incidentally, the 2 changes above also remove the need for
redefining the 400 response code multiple times in api.yaml, preventing
the validation errors when browsing the spec online

  - Fixed typo: "Successfull" -> "Successful"
  - Consistent double quoting of value strings in api.yaml
  - Consistent double quoting of format strings in
bootstrap.src/errors.py

### CODE REVIEW

File: bootstrap/src/errors.py [hunk 1]
```diff
@@ -53,16 +53,16 @@ def __init__(self, extra_param):
...
         message = f"Destination '{destination}' does not exist"
         super(DestinationDoesNotExistError, self).__init__(message)
 
 
 class LowBalanceError(BootstrapError):
     def __init__(self):
         self.code = 4003
-        self.http_code = 400
-        message = f'The account does not have enough kin to perform this operation'
+        self.http_code = 500
```
Comment (Ronserruya) [inline bootstrap/src/errors.py]: why was this changed? this doesn't match the api.yaml

**Review by Ronserruya — CHANGES_REQUESTED** [review]
Thanks for the changes! just added a small comment about a status code.

File: bootstrap/src/errors.py (hunk 1 above)
Comment (geebee) [reply bootstrap/src/errors.py]: I was originally debating making this a 500, since 4xx series codes imply to the client that there is something they can change about the request and retry with expected success and a low balance isn't really a client "mistake".

I ended up leaving it as is in `api.yaml` since you could reasonably say that the client can simply send less kin, and succeed on the request, and that also minimized externally facing changes.

I've pushed a commit (d9eca56bf25864bc58f2dff8ed48fe0d0849c99f) to the PR to re-align the python with the yaml.

**Comment by geebee** [comment]
@Ronserruya can I get a re-review please?

**Review by Ronserruya — APPROVED** [review]

**Comment by Ronserruya** [comment]
@geebee Thanks, since this is a major version change we will discuss internally when to merge it and release a new version

**Comment by geebee** [comment]
Since there doesn't seem to be a plan to release this soon, and I have further changes I'd like to merge to my fork's master branch without affecting this PR, I'm going to close it, and re-open another one from a new branch that is isolated to these changes. #14
